### PR TITLE
Add Jython support.

### PR DESCRIPTION
**Pull request by jamadden**
Unfortunately, 'tox' doesn't currently work with Jython. I set up travis to work with Jython as described at https://www.topbug.net/blog/2012/05/27/use-travis-ci-with-jython/ (referenced from http://docs.travis-ci.com/user/resources/).

### CODE REVIEW

**Comment by jamadden** [comment]
Per the discussion in zopefoundation/transaction#8, I removed the travis support, and unfortunately Tox doesn't work either. The tests are easy to run manually, though (and the commit that did have travis support ran green).

File: src/zc/lockfile/__init__.py [hunk 1]
```diff
@@ -27,10 +27,49 @@ class LockError(Exception):
...
+            import java.lang.Exception
+            import threading
+            import os.path
+        except ImportError:
+            def _lock_file(file):
+                raise TypeError('No file-locking support on this platform')
+            def _unlock_file(file):
+                raise TypeError('No file-locking support on this platform')
+        else:
+            # Jython Java file locks are per-VM, so they can't be used
```
Comment (jaraco) [inline src/zc/lockfile/__init__.py]: It took me several reads of this comment and the related code to get a grasp of what's happening. Let me see if I understand correctly - Java file locks only provide for synchronization across processes (VM instances), but for different threads within a process, any number of threads will acquire the same lock. Perhaps you could include text like that in the comment? Also, an authoritative reference to a description of this behavior would be nice (if available).

**Comment by jaraco** [comment]
The whole try/except/else blocks for _lock_file are already a little messy. I would say that following this change (if accepted), the platform-specific blocks should be refactored into encapsulated functions or modules.

File: src/zc/lockfile/__init__.py (hunk 1 above)
Comment (jamadden) [reply src/zc/lockfile/__init__.py]: Yes, that's what it means (and I thought what it said 😄 ). The authoritative link would be to the [tryLock javadoc](https://docs.oracle.com/javase/10/docs/api/java/nio/channels/FileChannel.html#tryLock(long,long,boolean)): 

> File locks are held on behalf of the entire Java virtual machine. They are not suitable for controlling access to a file by multiple threads within the same virtual machine.

**Comment by jamadden** [comment]
I no longer have interest in this PR (and in fact it seems the underlying repo fork got deleted  at some point). If someone wants to move it forward that's great, but unfortunately it won't be me. It can be kept open as a reminder or closed, either is fine with me.

**Pull request closed without merging**